numba/cuda/compiler.py

```diff
@@ -465,9 +467,15 @@ def _prepare_args(self, ty, val, stream, retr, kernelargs):
         Convert arguments to ctypes and append to kernelargs
         """
         if isinstance(ty, types.Array):
-            devary, conv = devicearray.auto_device(val, stream=stream)
-            if conv:
-                retr.append(lambda: devary.copy_to_host(val, stream=stream))
+            if isinstance(ty, types.NumbaArray):
```

Comment:
Should this be `SmartArray` or whatever it ends up being called for consistency? Same in boxing/typing.